website/blog/2020-06-16-jest-27.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Jest 27: Jest Architecture
+author: Kunal Kushwaha
+authorURL: http://twitter.com/kush_kunal
+authorFBID: 2627862387322202
+---
+## Why is Testing important
+Humans make mistakes all the time. When contributing to a large scale open source project, you are going to be working with other developers in the community. You need a way to test that the code you are going to contribute to the project, will not disrupt the existing functionality of the project. Software testing is necessary because it provides a way to check anything and everything we produce as things can always go wrong. It improves consistency and performance hence it is a crucial component of software product development. It also helps developers and testers to compare actual and expected results to improve quality. Testing is essential in providing reliable software that makes it safe to use in real-world applications.
+
+## What is Jest
+Jest is a JavaScript testing framework designed to ensure correctness of any JavaScript codebase. It allows you to write tests with an approachable, familiar and feature-rich API that gives you results quickly. Jest is well-documented, requires little configuration and can be extended to match your requirements. Jest makes testing delightful. 
+
+## Jest Architecture
+- Jest project is split up into multiple packages
+- jest-cli: most people install the ‘jest’ package which forwards to jest-cli. It pulls together all the other packages to run tests.
+- When you run $ jest my-test it goes to jest-cli. Jest-cli talks to jest-config.
+- Here’s how Jest works:
+
+### (STEP-1)
+- jest-config: It figures out what should happen, what the person wants to do on a cli environment. It takes the config from some configuration file (finds this file automatically) & it also takes the argument ( — flag arguments) that were made with the call and merges them together & figures out what we actually want to do.
+There is a function “normalize” in this package that normalizes the data that we get for this invocation and the it creates 2 separate configuration objects:
```

Comment:
There appears to be a typo in line 21 : "the it" @kunal-kushwaha

Reply:
@shivaylamba: Thanks for pointing it out!

Reply:
You are welcome!